mne/viz/epochs.py

```diff
@@ -65,71 +67,252 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
+        else:
+            raise ValueError("If `groupby` is not None, it must be a dict "
+                             "or 'type', got " + type(groupby))
+    return all_picks, all_ch_types, names  # all_picks is a list of lists
+
+
+def _get_to_plot(epochs, combine, all_picks, all_ch_types, scalings, names):
+    "Helper for plot_epochs_image. Packs data and metadata into a list."
+    data = epochs.get_data()
+    to_plot_list = list()
+
+    if combine is None:
+        for pick, ch_type, _ in zip(all_picks, all_ch_types, names):
+            this_data = data[:, pick, :]
+            to_plot_list.append((this_data, ch_type,
+                                 epochs.average(picks=[pick]),
+                                 epochs.ch_names[pick]))
+    else:
+        type2name = {"eeg": "EEG", "grad": "Gradiometers",
+                     "mag": "Magnetometers"}
+        if combine == "gfp":
+            combine = lambda data: np.sqrt((data * data).mean(axis=1))
+        elif combine == "mean":
+            combine = lambda data: np.mean(data, axis=1)
+        elif combine == "std":  # undocumented ...
+            combine = lambda data: np.std(data, axis=1)
+        elif combine == "median":  # undocumented ...
+            combine = lambda data: np.median(data, axis=1)
+        elif not callable(combine):
+            raise ValueError(
+                "`combine` must be None, a callable or one out of 'mean' "
+                "or 'gfp'. Got " + str(type(combine)))
+        for ch_type, picks_, name in zip(all_ch_types, all_picks, names):
+            if len(np.atleast_1d(picks_)) < 2:
+                raise ValueError("Cannot combine over only one sensor. "
+                                 "Consider using different values for "
+                                 "`picks` and/or `groupby`.")
+
+            this_data = combine(data[:, picks_, :])
+            evoked = epochs.average(picks=np.atleast_1d(picks_))
+            evoked.data = this_data.mean(0)[np.newaxis, :]
+            to_plot_list.append((this_data, ch_type, evoked,
+                                 type2name.get(name, name)))
+
+    return to_plot_list  # data, ch_type, evoked, name
+
+
+def _plot_epochs_image(data, sigma=0., vmin=None, vmax=None, colorbar=False,
+                       order=None, show=False, unit=None, cmap=None,
+                       fig=None, axes=None, overlay_times=None, scaling=None,
+                       evoked=None, title=None, draw_evoked=False):
+    """Helper function for plot_epochs_image/epochs.plot_image."""
+    from scipy import ndimage
+    import matplotlib.pyplot as plt
+
+    ### prepare fig and axes ###
     if axes is not None:
         if fig is not None:
             raise ValueError('Both figure and axes were passed, please'
                              'decide between the two.')
         from .utils import _validate_if_list_of_axes
-        oblig_len = 3 if colorbar else 2
+        oblig_len = 3
```

Comment:
you could also do `oblig_len = 3 - ((not colorbar) + (not draw_evoked))` i you don't think it is too convoluted.